upgrade_tests/upgrade_schema_agreement_test.py

```diff
@@ -0,0 +1,258 @@
+import re
+import time
+
+from ccmlib.node import TimeoutError
+from dtest import Tester, debug
+
+
+class TestSchemaAgreementUpgrade(Tester):
+    """
+    Verifies that upgrades do not produce migration storms.
+
+    For upgrades from 3.0 to 3.x we have encountered that the beloved 'cdc' column
+    caused rolling upgrades to produce migration storms, because the schema digest calculated
+    by 3.11 differs from that in 3.0, because 3.11 included the 'cdc' column in the digest
+    calculation.
+
+    This test verifies that (after some grace time, i.e. nodes in status NORMAL) neither the
+    upgraded node nor the other nodes perform schema pulls (migration tasks) resulting in a
+    "migration storm".
+
+    A few schema migrations however happen before all nodes are in status NORMAL. This is not ideal,
+    but legit. In an ideal world, those should not happen - but we would have to do the schema
+    changes atomically - and currently there is just nothing like a concurrent schema change.
+
+    The tests run for a quite some time, about 5 minutes per test.
+    But there's some debug output for your entertainment - enjoy.
+
+    The node.start/stop calls pass wait_other_notice=False intentionally, because there seem to
+    be issues in 3.0 to thoroughly detect and _log_ the status change - i.e. the wait_other_notice
+    check doesn't seem to be reliable - at least not in this test.
+    """
+
+    # The number of seconds we wait for schema migration log entries to verify
+    migration_check_time = 30
+
+    def __init__(self, *args, **kwargs):
```

Comment:
this needs to be in a fixture now... these log pattern whitelists won't get pulled in.. see bootstrap_test.py fixture_add_additional_log_patterns for an example.